### PR TITLE
State: Add action creators, data layer and reducer for user profile links

### PR DESCRIPTION
**Pull request by tyxla**
This PR introduces action creators, data layer and reducer for fetching, adding and deleting user profile links. All of those are accompanied by their respective tests.

Part of #20241. 

To test:
* Checkout this branch
* Verify all tests pass: `npm run test-client user-profile-links`

### CODE REVIEW

**Comment by sirreal** [comment]
The tests pass according to testing instructions. I'll try to give a more detailed review soon.

**Comment by seear** [comment]
A general point—the [convention in the data layer](https://github.com/Automattic/wp-calypso/tree/master/client/state/data-layer#file-structure) is for the file structure to mirror the API URL path, so the handler for requesting user profile links, for which the API is `/me/settings/profile-links`, would live at `client/state/data-layer/wpcom/me/settings/user-profile-links/index.js`.

File: client/state/user-profile-links/reducer.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+} from 'state/action-types';
+
+export const items = createReducer( [], {
+	[ USER_PROFILE_LINKS_RECEIVE ]: ( state, { profileLinks } ) => profileLinks,
+	[ USER_PROFILE_LINKS_DELETE_SUCCESS ]: ( state, { linkSlug } ) =>
+		reject( state, { link_slug: linkSlug } ),
+} );
+
+export const errors = createReducer(
+	{},
```
Comment (seear) [inline client/state/user-profile-links/reducer.js]: Depending on how you use the errors you may not need this reducer. Notices can be shown from the data layer handlers or from the dedicated [notice middleware](https://github.com/Automattic/wp-calypso/blob/master/client/state/notices/middleware.js).

Comment (tyxla) [reply client/state/user-profile-links/reducer.js]: Yes, good call. Actually, that was the initial plan, but then I saw that we don't use global notices for user profile links - we rather display them inline:





So we need those errors to be stored in the Redux state tree - I don't really want to introduce any behavioral/visual changes with this refactor.

Comment (seear) [reply client/state/user-profile-links/reducer.js]: 👍

**Comment by tyxla** [comment]
Thanks for the review @sirreal and @seear!

> It would be nice to use dispatchRequestEx, which I believe will be replacing dispatchRequest:

Done in 56b7ddd83902b61cb6f0f6ccf3234ff20fb2cbab.

> Maybe just want to alter the file structure to match the API.

Done in a00a1408ee590e6a3eecdf2a1766f409952cddc1 and fbfd05b6e0ef8881a5d8b375c1d98e90b3a3c38a.

Could you have another 👀  before I 🚢 ? Thanks!

File: client/state/data-layer/wpcom/me/settings/profile-links/index.js [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+ * @param   {Object}   action   Redux action
+ * @returns {Object} Dispatched http action
+ */
+export const addUserProfileLinks = ( { dispatch }, action ) =>
+	dispatch(
+		http(
+			{
+				apiVersion: '1.2',
+				method: 'POST',
+				path: '/me/settings/profile-links/new',
```
Comment (seear) [inline client/state/data-layer/wpcom/me/settings/profile-links/index.js]: The idea is that this dispatch would go in a separate module at `/me/settings/profile-links/new/index.js`, see this [example](https://github.com/Automattic/wp-calypso/tree/master/client/state/data-layer/wpcom/activity-log/rewind).

Comment (tyxla) [reply client/state/data-layer/wpcom/me/settings/profile-links/index.js]: Ah, makes sense - applied in bacd559af4acca4e4de1493bc0dec858b3d4b54c - I love that structure, thanks! ❤️

**Comment by tyxla** [comment]
Feedback addressed, this one is ready for a new 👁

**Comment by tyxla** [comment]
Thank you both 🙇

**Pull request merged**